### PR TITLE
Add endpoint to get all bookings for event made by user

### PR DESCRIPTION
**Pull request by bh412**
- Returns all users booked onto an event in your groups

### CODE REVIEW

**Review by Morpheu5 — APPROVED** [review]

File: src/main/java/uk/ac/cam/cl/dtg/isaac/api/EventsFacade.java [hunk 1]
```diff
@@ -502,12 +506,11 @@ public final Response getEventBookingByEventId(@Context final HttpServletRequest
...
             RegisteredUserDTO currentUser = userManager.getCurrentRegisteredUser(request);
 
-            List<EventBookingDTO> eventBookings = bookingManager.getBookingsByEventId(eventId);
+            List<EventBookingDTO> eventBookings = this.mapper.mapAsList(bookingManager.getBookingsByEventId(eventId), EventBookingDTO.class);
 
             // Event leaders are only allowed to see the bookings of connected users
-            if (Role.EVENT_LEADER.equals(currentUser.getRole())) {
-                eventBookings = userAssociationManager.filterUnassociatedRecords(
-                        currentUser, eventBookings, booking -> booking.getUserBooked().getId());
+            if (Arrays.asList(Role.STUDENT, Role.TEACHER, Role.EVENT_LEADER, Role.CONTENT_EDITOR).contains(currentUser.getRole())) {
```
Comment (bh412) [inline src/main/java/uk/ac/cam/cl/dtg/isaac/api/EventsFacade.java:512]: Pretty sure this shouldn't contain Role.STUDENT?

**Review by bh412 — COMMENTED** [review]
Code looks ok aside from the flagged issue - testing now

File: src/main/java/uk/ac/cam/cl/dtg/isaac/api/EventsFacade.java (hunk 1 above)
Comment (Morpheu5) [reply src/main/java/uk/ac/cam/cl/dtg/isaac/api/EventsFacade.java]: Likely not. This wouldn't prevent you from seeing your own bookings now, would it?

File: src/main/java/uk/ac/cam/cl/dtg/isaac/api/EventsFacade.java [hunk 2]
```diff
@@ -582,6 +584,40 @@ public final Response getEventBookingForGivenGroup(@Context final HttpServletReq
...
+    @GZIP
+    @ApiOperation(value = "List event bookings for a specific event")
+    public final Response getEventBookingForAllGroups(@Context final HttpServletRequest request,
+                                                       @PathParam("event_id") final String eventId) {
+        try {
+            RegisteredUserDTO currentUser = userManager.getCurrentRegisteredUser(request);
+            List<EventBookingDTO> eventBookings = this.mapper.mapAsList(bookingManager.getBookingsByEventId(eventId), EventBookingDTO.class);
+
+            if (!isUserTeacherOrAbove(userManager, currentUser)) {
+                return new SegueErrorResponse(Status.FORBIDDEN, "You do not have permission to use this endpoint.").toResponse();
```
Comment (jsharkey13) [inline src/main/java/uk/ac/cam/cl/dtg/isaac/api/EventsFacade.java:602]: There's already `SegueErrorResponse.getIncorrectRoleResponse()` (or something) for this. Not sure it's worth changing this now, but for reference!

**Review by jsharkey13 — APPROVED** [review]
I think this all looks sensible, if it has been tested!

**Pull request merged**